### PR TITLE
add ability to fill between lines

### PR DESCRIPTION
**Pull request by gidden**
# Please confirm that this PR has done the following:

- [x] Tests Added
- [x] Documentation Added
- [x] Description in RELEASE_NOTES.md Added

## Adding to RELEASE_NOTES.md (remove section after adding to RELEASE_NOTES.md)

Please add a single line in the release notes similar to the following:

```
- (#XX)[http://link-to-pr.com] Added feature which does something
```

# Description of PR

Please describe the changes introduced by this PR.

### CODE REVIEW

**Comment by gidden** [comment]
This is good to go pending CI, @danielhuppmann

**Comment by gidden** [comment]
ok, now actually g2g @danielhuppmann

**Comment by gidden** [comment]
hold on this, testing with the SR1.5 dataset

**Comment by gidden** [comment]
Ok, found some other issues which are now fixed and tested. Finally ready, @danielhuppmann

**Comment by danielhuppmann** [comment]
what can I say other than - amazing!

File: tests/test_plotting.py [hunk 1]
```diff
@@ -70,6 +71,39 @@ def test_line_color(plot_df):
...
+    newdata = ['test_model1', 'test_scenario1', 'World', 'Primary Energy|Coal',
+               'EJ/y', 2010, 3.50]
+    df.data.loc[len(df.data) - 1] = newdata
+    newdata = ['test_model1', 'test_scenario1', 'World', 'Primary Energy|Coal',
+               'EJ/y', 2012, 3.50]
+    df.data.loc[len(df.data)] = newdata
+    newdata = ['test_model1', 'test_scenario1', 'World', 'Primary Energy|Coal',
+               'EJ/y', 2015, 3.50]
+    df.data.loc[len(df.data) + 1] = newdata
+    print(df.data)
```
Comment (danielhuppmann) [inline tests/test_plotting.py]: please remove `print`

**Comment by danielhuppmann** [comment]
apology accepted - very glad that this doesn't only happens to me...

**Pull request merged**